WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
One of the many legendary problems in the `java.util.Date` class is that the months are 0 based. January is0, December is 11.

Maybe this off by one design is inevitable, but it's an error magnet.